Mathlib/Order/Interval.lean

```diff
@@ -731,22 +739,43 @@ noncomputable instance completeLattice [@DecidableRel α (· ≤ ·)] :
             let f := fun (s : NonemptyInterval α) (ha : ↑s ∈ s₁) => s.toProd.fst
             exact WithBot.coe_le_coe.2 ⟨le_supᵢ₂ (f := f) s ha, infᵢ₂_le s ha⟩
           · exact bot_le
-        le_infₛ := fun S s ha => by
+        le_infₛ := by
+          intro S s ha
           cases s with
           | none => exact bot_le
           | some s =>
             dsimp only -- Porting note: added
             split_ifs with h
-            · exact
+            . exact
                 WithBot.some_le_some.2
                   ⟨supᵢ₂_le fun t hb => (WithBot.coe_le_coe.1 <| ha _ hb).1,
                     le_infᵢ₂ fun t hb => (WithBot.coe_le_coe.1 <| ha _ hb).2⟩
-            rw [not_and_or, Classical.not_not] at h
-            rcases h with h | h
-            · exact ha _ h
-            cases h fun t hb c hc =>
-                (WithBot.coe_le_coe.1 <| ha _ hb).1.trans <|
-                  s.fst_le_snd.trans (WithBot.coe_le_coe.1 <| ha _ hc).2 }
+            · rw [not_and_or, not_not] at h
+              rcases h with h | h
+              · exact ha _ h
+              · -- Porting note: This part has been redone, since the original
+                -- mathlib3 proof had problems with goint from `toProd` to
+                -- `toDualProd`. Original mathport output:
+                -- cases h fun t hb c hc =>
+                --   (WithBot.coe_le_coe.1 <| ha _ hb).1.trans <|
+                --     s.fst_le_snd.trans (WithBot.coe_le_coe.1 <| ha _ hc).2 }
+                exfalso
+                apply h
+                intro b hb c hc
+                have h₁ := (WithBot.coe_le_coe.1 <| ha _ hb).1
+                have h₂ := (WithBot.coe_le_coe.1 <| ha _ hc).2
+                have h := (s.fst_le_snd.trans h₂)
+                clear h₂
+                have h₃ : b.toProd.fst ≤ s.toProd.fst
+                · repeat rw [NonemptyInterval.toDualProd_apply] at h₁
+                  dsimp at h₁
+                  -- Porting note: (TODO): Why does this work the way it does.
+                  -- This next proof step looks somehow very sketchy to me!
+                  -- I beleive the API around duals might hav a problem.
+                  rw [OrderDual.le_def] at h₁
```

Comment:
This seems somewhat sketchy. I think something about the `OrderDual` API does not work as intended.

This step essentailly turns `toDual s.toProd.fst ≤ toDual b.toProd.fst` into `toDual b.toProd.fst ≤ toDual s.toProd.fst` and it is not obvious what hidden instances it changes in order to do so.

Reply:
short discussion about this on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/!4.233360.20OrderDual).